evalml/pipelines/pipeline_base.py

```diff
@@ -313,20 +310,11 @@ def hyperparameters(cls):
         """Returns hyperparameter ranges from all components as a dictionary"""
         hyperparameter_ranges = dict()
         component_graph = copy.copy(cls.component_graph)
-        if isinstance(component_graph, list):
-            for component_class in component_graph:
-                component_class = handle_component_class(component_class)
-                component_hyperparameters = copy.copy(component_class.hyperparameter_ranges)
-                if cls.custom_hyperparameters and component_class.name in cls.custom_hyperparameters:
-                    component_hyperparameters.update(cls.custom_hyperparameters.get(component_class.name, {}))
-                hyperparameter_ranges[component_class.name] = component_hyperparameters
-        else:
-            for component_name, component_info in component_graph.items():
-                component_class = handle_component_class(component_info[0])
-                component_hyperparameters = copy.copy(component_class.hyperparameter_ranges)
-                if cls.custom_hyperparameters and component_name in cls.custom_hyperparameters:
-                    component_hyperparameters.update(cls.custom_hyperparameters.get(component_name, {}))
-                hyperparameter_ranges[component_name] = component_hyperparameters
+        for component_name, component_class in ComponentGraph.linearized_component_graph(component_graph):
```

Comment:
JW, why not call for component_name, component_class in PipelineBase.linearized_component_graph?

Reply:
Just cause I saw the existing implementation used a copy of the component graph lol. I don't think that's necessary though? In that case, I can just use the pipeline class property 👍